Rust Community Update

soothethesenses opened the pull request:
- added links to YouTube channel,  meetup page and embedded playlist
- site now has layout and design
- Rust logo added

soothethesenses committed:
Rust Community Update

- added links to YouTube channel,  meetup page and embedded playlist
- site now has layout and design
- Rust logo added

No9 commented on .eleventy.js:
```diff
@@ -0,0 +1,25 @@
+	
+// adds embed plugin
+config.addPlugin(embedYouTube);
+
```
the `css` folder needs to move into the `src` folder and `config.addPassthroughCopy('src/css');` needs to be added here

No9 replied:
Actually it looks like the css folder can be deleted and a new one added to map the fonts

No9 commented on build/_site/index.html:
```diff
@@ -0,0 +1,35 @@
+<!DOCTYPE html>
```
This file and the `build/_site` folder needs deleting

No9 commented on src/index.md:
```diff
@@ -0,0 +1,12 @@
+layout: page.njk
+---
+
+<link rel="stylesheet" href="C:\Users\Cillian\Documents\GitHub\website\src\fonts" type="text/css" charset="utf-8" />
```
This shouldn't be a hard link to a file on your machine. You should create a file in `src/css/stylesheet.css` 

And add the following content to it
```
@font-face {
   font-family: alfa_slab;
   src: url(/fonts/alfaslab/alfaslabone-regular-webfont.woff);
}
@font-face {
   font-family: fira_sans;
   src: url(/fonts/firasans/firasans-regular-webfont.woff);
}
```
You should then update your font-families in the tags to match the ones defined above.

No9 commented on build/fonts/firasans/firasans-regular-demo/index.html:
```diff
@@ -0,0 +1,622 @@
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
```
This sample html documents need to be ignored.
In the root of this project create a file called `.eleventyignore` and add the paths to the sample pages to it.
e.g, `build/fonts/firasans/firasans-regular-demo/index.html`

soothethesenses committed:
Commit

- css folder deleted
- build/_site and src/_site deleted
- src/css stylesheet added
- removed fonts folder

No9 commented:
Add fonts folder back in
update the attributes in the partials to reference the font-family in css and point the stylesheet to /css/stylesheet.css
remove css/components and css/main.scss

soothethesenses committed:
Commit 3

removed compoments and main.scss in css
updated partials attributes
readded fonts folder

No9 commented:
Add line to .eleventy.js to copy css folder
Update styles in `_includes/page.njk to use `alfa_slab` and `fira_sans`
Update `includes/partials/htmlhead.njk` to point to the css/stylesheet.
` <link rel="stylesheet" href="/css/stylesheet.css" type="text/css" charset="utf-8" />`
Update the body style to use `alfa_slab`

soothethesenses committed:
4...

- css folder copies into build
- updated styles in _includes

soothethesenses committed:
Pipeline

pipeline added to readme.md to explain what happens when you type npx eleventy --serve

No9 commented on README.md:
```diff
@@ -5,6 +5,15 @@ The public facing website for rust dublin
 In order to build this site, run the follow commands in a terminal.
 Pull requests should be made to the public branch
 
+## pipeline
```
In general the site is generated by processing the `src` input folder defined in `.eleventy.js` and outputting it to the `build` directory
2. It takes the `.md` files in the `src` folder merges them with page.njk and places the output in the `build` directory

Remove the reference to ~~using the template provided~~
~~this site outputs to build - this is controlled by .eleventy.js.~~

soothethesenses committed:
README.MD update #2
